### PR TITLE
6595 volumeinfo generate

### PR DESCRIPTION
**Pull request by blackpiglet**
Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: pkg/controller/backup_controller.go [hunk 1]
```diff
@@ -908,3 +911,316 @@ func oldAndNewFilterParametersUsedTogether(backupSpec velerov1api.BackupSpec) bo
...
+		}
+		if pvcPVInfo, ok := backup.PVMap[volumeSnapshot.Namespace+"/"+*volumeSnapshot.Spec.Source.PersistentVolumeClaimName]; ok {
+			volumeInfo := volume.VolumeInfo{
+				BackupMethod:      volume.CSISnapshot,
+				PVCName:           pvcPVInfo.PVCName,
+				PVCNamespace:      pvcPVInfo.PVCNamespace,
+				PVName:            pvcPVInfo.PV.Name,
+				Skipped:           false,
+				SnapshotDataMoved: false,
+				CSISnapshotInfo: volume.CSISnapshotInfo{
```
[the remark this answers is not in the document]
Comment (blackpiglet) [reply pkg/controller/backup_controller.go:1084]: Added, and the comment is modified.

File: pkg/controller/backup_controller.go [hunk 2]
```diff
@@ -908,3 +911,316 @@ func oldAndNewFilterParametersUsedTogether(backupSpec velerov1api.BackupSpec) bo
...
+					SnapshotDataMoved: true,
+					Skipped:           false,
+					CSISnapshotInfo: volume.CSISnapshotInfo{
+						Driver: driverUsedByVSClass,
+					},
+					SnapshotDataMovementInfo: volume.SnapshotDataMovementInfo{
+						DataMover:        "velero",
+						UploaderType:     "kopia",
+						RetainedSnapshot: dataUpload.Namespace + "/" + dataUpload.Spec.CSISnapshot.VolumeSnapshot,
+						SnapshotHandle:   dataUpload.Status.SnapshotID,
```
[the remark this answers is not in the document]
Comment (blackpiglet) [reply pkg/controller/backup_controller.go]: Removed.

File: pkg/controller/backup_controller.go [hunk 3]
```diff
@@ -908,3 +911,316 @@ func oldAndNewFilterParametersUsedTogether(backupSpec velerov1api.BackupSpec) bo
...
+					PVCNamespace:      pvcPVInfo.PVCNamespace,
+					PVName:            pvcPVInfo.PV.Name,
+					SnapshotDataMoved: true,
+					Skipped:           false,
+					CSISnapshotInfo: volume.CSISnapshotInfo{
+						Driver: driverUsedByVSClass,
+					},
+					SnapshotDataMovementInfo: volume.SnapshotDataMovementInfo{
+						DataMover:        "velero",
+						UploaderType:     "kopia",
```
[the remark this answers is not in the document]
Comment (blackpiglet) [reply pkg/controller/backup_controller.go]: DataMover is modified. UploaderType is not stored in the DUCR yet, so hard-code as "kopia" for now.

File: pkg/controller/backup_controller.go [hunk 4]
```diff
@@ -908,3 +911,316 @@ func oldAndNewFilterParametersUsedTogether(backupSpec velerov1api.BackupSpec) bo
...
+						RetainedSnapshot: dataUpload.Namespace + "/" + dataUpload.Spec.CSISnapshot.VolumeSnapshot,
```
[the remark this answers is not in the document]
Comment (blackpiglet) [reply pkg/controller/backup_controller.go]: Removed.

**Review by Lyndon-Li — APPROVED** [review]

**Review by ywk253100 — APPROVED** [review]

**Pull request merged**